include/swift/AST/Attr.def

```diff
@@ -121,11 +121,12 @@ DECL_ATTR(available, Available,
   1)
 CONTEXTUAL_SIMPLE_DECL_ATTR(final, Final,
   OnClass | OnFunc | OnAccessor | OnVar | OnSubscript |
-  DeclModifier,
+  DeclModifier | ABIBreakingToAdd | ABIBreakingToRemove |
+  APIBreakingToAdd | APIBreakingToRemove,
```

Comment:
Now that we have this flexibility, we can note that `final` is not `APIBreakingToRemove`.